pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/SegmentPreProcessor.java

```diff
@@ -130,7 +130,24 @@ public void process(@Nullable SegmentOperationsThrottler segmentOperationsThrott
         if (type != StandardIndexes.forward()) {
           IndexHandler handler = createHandler(type);
           indexHandlers.add(handler);
-          handler.updateIndices(segmentWriter);
+          try {
+            handler.updateIndices(segmentWriter);
+          } catch (IllegalStateException | UnsupportedOperationException e) {
```

Comment:
IMO we don't want to differentiate the handling of different exceptions. We are not strictly following a convention for exception type, e.g. why not catching `IllegalArgumentException`? We should just use a flag to control whether to continue on error to make the behavior more deterministic.

Reply:
Agreed,  I would say then we should define an exception type that allow us to continueOnError by default.

I catch these 2 exceptions because only these two exceptions would throw for now.